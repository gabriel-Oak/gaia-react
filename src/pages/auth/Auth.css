.Auth form > .field_wrapper {
  display: flex;
  flex-direction: column;
}