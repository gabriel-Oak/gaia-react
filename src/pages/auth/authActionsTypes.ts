export interface AuthAction {
    type: string;
    value?: any;
}

export const AuthActionsTypes = {

}
