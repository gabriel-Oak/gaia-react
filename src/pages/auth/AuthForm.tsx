import React from 'react'

import { Field, reduxForm } from 'redux-form';
import { required } from '../../shared/validators/validators';
import RenderTextField from '../../shared/components/renderTextField/renderTextField';

import { Button, CardActions, CardContent, CircularProgress } from '@material-ui/core';

import './Auth.css';

const AuthForm = (props: any) => {
  const { handleSubmit, onSubmit, loading } = props

  return (
    <form onSubmit={handleSubmit(onSubmit)}>

      <CardContent className="field_wrapper">
        <Field
          name="user"
          label="UsuÃ¡rio"
          variant="outlined"
          disabled={loading}
          component={RenderTextField}
          validate={[required]}
        />

        <Field
          name="senha"
          label="Senha"
          variant="outlined"
          disabled={loading}
          component={RenderTextField}
          validate={[required]}
        />
      </CardContent>

      <CardActions className="button-container">
        <Button
          color="primary"
          variant="contained"
          type="submit"
          disabled={loading}
        >
          {
            loading
              ? <CircularProgress size={24} />
              : 'Entrar'
          }
        </Button>
      </CardActions>
    </form>
  );
}

export default reduxForm({ form: 'auth' })(AuthForm);
