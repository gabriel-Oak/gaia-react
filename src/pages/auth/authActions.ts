import axios from 'axios';

import { Dispatch } from "redux";
import { AuthActionsTypes } from "./authActionsTypes";
import { api } from '../../enviroments/enviroments';
import resolveError from '../../shared/utils/resolveError';

export const logIn = (form: any) => async (dispatch: Dispatch) => {
  try {

    dispatch({ type: AuthActionsTypes.SENDED });

    const { data } = await axios.post(`${api.user}/${form.user}`, form);

    console.log(data);


  } catch (e) {
    dispatch({
      type: AuthActionsTypes.FIRE_SNACK,
      value: {
        open: true,
        type: 'error',
        message: resolveError(e).message,
        duration: 6000
      }
    })
  } finally {
    dispatch({ type: AuthActionsTypes.ENDED });
  }
}

export const closeSnack = () => ({
  type: AuthActionsTypes.CLOSE_SNACK
});