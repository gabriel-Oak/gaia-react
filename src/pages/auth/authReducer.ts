import { AuthAction, AuthActionsTypes } from './authActionsTypes';

export interface AuthState {
    loading: boolean;
    auth: boolean;
    snackbar: {
        open: boolean;
        message?: string;
        type?: 'success' | 'error' | 'info' | 'warning';
        duration?: number
    };
}

const INITIAL_STATE: AuthState = {
    loading: false,
    auth: false,
    snackbar: {
        open: false
    }
}

const authReducer = (state = INITIAL_STATE, action: AuthAction) => {
    switch (action.type) {
        case AuthActionsTypes.SENDED:
            return { ...state, loading: true }

        case AuthActionsTypes.ENDED:
            return { ...state, loading: false }

        case AuthActionsTypes.SUCCESS:
            return { ...state, auth: true }

        case AuthActionsTypes.FIRE_SNACK:
            return { ...state, snackbar: action.value }

        case AuthActionsTypes.CLOSE_SNACK:
            return { ...state, snackbar: { open: false } }

        default:
            return state;
    }
}

export default authReducer;
