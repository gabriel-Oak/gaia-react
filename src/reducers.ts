import { combineReducers } from 'redux';

import authReducer from './pages/auth/authReducer';
import mainReducer from './pages/main/mainReducer';

export interface ReducersPool {
  mainReducer: object;
  authReducer: object;
}

export default combineReducers({
  authReducer,
  mainReducer
});