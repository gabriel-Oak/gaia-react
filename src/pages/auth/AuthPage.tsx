import React, { PureComponent, ReactNode } from 'react'
import { ReducersPool } from '../../reducers';
import { AuthState } from './authReducer';
import { connect } from 'react-redux';

import { setTitle } from '../main/mainActions';

import { Card, CardHeader } from '@material-ui/core';
import AuthForm from './AuthForm';

interface Props extends AuthState {
  setTitle: Function;
}

class AuthPage extends PureComponent<Props> {
  constructor(props: Props) {
    super(props);

    this.props.setTitle('Login');
  }

  render(): ReactNode {
    return (
      <main className="Auth">
        <article className="Center-container s600">
          <Card>
            <CardHeader title="Login" />
            <AuthForm />
          </Card>
        </article>
      </main>
    )
  }
}

const mapStateToProps = (state: ReducersPool) => {
  const { authReducer } = state;
  return authReducer;
};

const mapDispatchToProps = {
  setTitle
};

export default connect(mapStateToProps, mapDispatchToProps)(AuthPage);
