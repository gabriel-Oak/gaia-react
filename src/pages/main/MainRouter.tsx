import React, { FC } from 'react'
import { Switch, Route, Redirect, } from 'react-router-dom'
import { fetchUser, closeSnack, toggleDrawer, redirect_to } from './mainActions';

import HomePage from './Home/HomePage'
import { connect } from 'react-redux';
import { ReducersPool } from '../../reducers';
import { mainState } from './mainReducer';
import { getSession } from '../../shared/utils/auth';

import Snack from '../../shared/components/snack/snack';
import { CircularProgress, Container } from '@material-ui/core';
import ToolBar from './components/ToolBar';
import { History } from 'history';

import SideDrawer from './components/SideDrawer';
import NotFoundPage from './NotFound/NotFoundPage';
import { historyTesting } from '../../shared/testMocks/history';
import UserFormContainer from './Users/Form/UserFormContainer';
import ProfileContainer from './Profile/ProfileContainer';

import useStyles from './styles';
import './Main.css';
import UserListContainer from './Users/List/UserListContainer';
import MenusFormContainer from './Menus/Form/MenusFormContainer';
import ExchangesContainer from './Exchanges/ExchangesContainer';

interface Props extends mainState {
  fetchUser: Function;
  closeSnack: Function;
  toggleDrawer: Function;
  redirect_to: Function;
  history: History | historyTesting;
  store?: any;
}


const MainRouter: FC<Props> = (props: Props) => {

  const classes = useStyles();
  const {
    snackbar,
    closeSnack,
    user,
    fetchUser,
    title,
    drawer,
    toggleDrawer,
    redirect,
    redirect_to,
    history
  } = props;

  const session = getSession();

  if (session.expired) {
    return (<Redirect to="login" />)
  }

  if (!user) {
    fetchUser(session.token, history);
  }

  if (redirect) {
    redirect_to(null);
    return (<Redirect to={redirect} />)
  }

  return (
    <div className="Main">
      <ToolBar
        title={title}
        history={history}
        toggleDrawer={toggleDrawer}
      />

      <SideDrawer
        open={drawer}
        redirect_to={redirect_to}
        toggleDrawer={toggleDrawer}
        user={user || { admin: false, }}
      />

      {
        user
          ?
          <Container
            maxWidth="md"
            className={classes.container}
          >
            <Switch>
              <Route path="/" exact component={HomePage} />
              {user.admin && (
                <>
                  <Route path="/users" exact component={UserListContainer} />
                  <Route path="/users/new" exact component={UserFormContainer} />
                  <Route path="/users/:userId/edit" exact component={UserFormContainer} />
                  <Route path="/profile/edit" exact component={ProfileContainer} />
                  <Route path="/menus/" exact component={MenusFormContainer} />
                  <Route path="/trocas" exact component={ExchangesContainer} />
                </>
              )}
              <Route path="**" component={NotFoundPage} />
            </Switch>
          </Container>
          :
          <div className="Center-container">
            <CircularProgress size={80} />
          </div>
      }
      <Snack
        closeSnack={closeSnack}
        {...snackbar}
      />
    </div>
  );
}

const mapStateToProps = (state: ReducersPool) => {
  const { mainReducer } = state;
  return mainReducer;
};

const mapDispatchToProps = {
  fetchUser,
  closeSnack,
  toggleDrawer,
  redirect_to
};

export default connect(mapStateToProps, mapDispatchToProps)(MainRouter);