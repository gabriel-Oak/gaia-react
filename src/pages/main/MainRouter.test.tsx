import React from 'react';
import ReactDOM from 'react-dom';

import { createStore, applyMiddleware, combineReducers } from 'redux';
import reducers from '../../reducers';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import MainRouter from './MainRouter';
import { BrowserRouter } from 'react-router-dom';
import { clearSession, setSession } from '../../shared/utils/auth';
import { configureShallow } from '../../shared/utils/testRender';

import { historyMock } from '../../shared/testMocks/history';

describe('MainRouter', () => {
  let store: any;
  let mockedStore: any;

  const redirectStore = configureStore([thunk])({
    mainReducer: {
      user: true,
      redirect: 'test'
    }
  });

  beforeEach(() => {
    store = createStore(
      reducers,
      applyMiddleware(thunk)
    );

    mockedStore = configureStore([thunk])({
      mainReducer: {
        title: 'Gaia',
        drawer: false,
        snackbar: {
          open: false,
        },
        user: {
          admin: true,
          email: 'test@email.cone',
          nome: 'Jorginho',
          user: 'string'
        },
      }
    });
  });

  afterEach(() => {
    clearSession();
  });

  it('redirect to login', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <Provider store={store}>
        <BrowserRouter>
          <MainRouter history={historyMock} />
        </BrowserRouter>
      </Provider>
      , div
    );

    expect(div.querySelector('.Main')).toBeFalsy();
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders without crashing', () => {
    setSession('yolooooooo');
    const div = document.createElement('div');

    ReactDOM.render(
      <Provider store={mockedStore}>
        <BrowserRouter>
          <MainRouter history={historyMock} />
        </BrowserRouter>
      </Provider>
      , div
    );

    expect(div.querySelector('.Main')).toBeTruthy();
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    setSession('yolooooooo');

    const page = configureShallow(
      <MainRouter history={historyMock} store={mockedStore} />
    );

    expect(page).toMatchSnapshot();
  });

  it('should try to fetch user data', () => {
    setSession('yolooooooo');

    const page = configureShallow(
      <MainRouter history={historyMock} store={store} />
    );
    expect(page.find('.Main')).toBeTruthy();
  });

  it('should perform a redirect', () => {
    setSession('yolooooooo');
    

    const page = configureShallow(
      <MainRouter history={historyMock} store={redirectStore} />
    );
    expect(page.find('.Main')).toBeTruthy();
  });
});