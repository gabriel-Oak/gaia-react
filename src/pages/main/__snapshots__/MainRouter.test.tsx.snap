// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainRouter should match snapshot 1`] = `
<div
  className="Main"
>
  <ToolBar
    history={
      Object {
        "goBack": [Function],
        "push": [Function],
      }
    }
    title="Gaia"
    toggleDrawer={[Function]}
  />
  <SideDrawer
    open={false}
    redirect_to={[Function]}
    toggleDrawer={[Function]}
  />
  <BrowserRouter>
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(HomePage)",
            "type": [Function],
          }
        }
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(NotFoundPage)",
            "type": [Function],
          }
        }
        path="**"
      />
    </Switch>
  </BrowserRouter>
  <Snack
    closeSnack={[Function]}
    open={false}
  />
</div>
`;
